### PR TITLE
Add `dryrun` query param to enable verification preview

### LINKED ISSUE #1319: Support for verification preview
Our verification UI flow would need to call the `/session/verify-checked` and `/session/input-files` API endpoints without actually persisting the result of the verification in the repository. This is required in order to obtain an exact preview of what the result of the verification is going to be for a given contract (from https://github.com/hashgraph/hedera-sourcify/issues/90)

Here's our implementation by introducing a `dryrun` query parameter
- https://github.com/hashgraph/hedera-sourcify/pull/92

Would you consider including this feature?

### PR DESCRIPTION
**Pull request by acuarica**
This PR adds a new query parameters to enable verification preview.

Fixes #1319.

### CODE REVIEW

**Comment by acuarica** [comment]
Hi @kuzdogan @marcocastignoli, I need to yet include tests for this feature. What is the test flow to make this ready for review?

Running `npm run test-local` fails with a bunch of errors, is that expected?

**Comment by marcocastignoli** [comment]
> Hi @kuzdogan @marcocastignoli, I need to yet include tests for this feature. What is the test flow to make this ready for review?
> 
> Running `npm run test-local` fails with a bunch of errors, is that expected?

I'll take car of it, thanks for the PR 👍

**Comment by marcocastignoli** [comment]
- [ ] Add logger.info for dryRun

**Comment by acuarica** [comment]
> I'll take car of it, thanks for the PR

Great, thanks! Let me know if there is something I could do to merge this. Happy to contribute here!

File: services/server/src/server/controllers/verification/verification.common.ts [hunk 1]
```diff
@@ -455,6 +456,12 @@ export const verifyContractsInSession = async (
     contractWrapper.status = getMatchStatus(match) || "error";
     contractWrapper.statusMessage = match.message;
     contractWrapper.storageTimestamp = match.storageTimestamp;
+    if (dryRun) {
+      logger.info("dryRun verification", {
+        sessionId: session.id,
```
Comment (marcocastignoli) [inline services/server/src/server/controllers/verification/verification.common.ts:461]: @kuzdogan for session cases, is it enough to log only the sessionId? All the other information are available in previous logs, right?

**Comment by kuzdogan** [comment]
@acuarica Do you only need this in the session endpoint? Because your changes only touch that enpoint and also does not change the UI

File: services/server/src/server/controllers/verification/verification.common.ts (hunk 1 above)
Comment (kuzdogan) [reply services/server/src/server/controllers/verification/verification.common.ts:461]: Hmm it doesn't seem to be the case. You can see by running it once. But generally just log more things than less

**Comment by acuarica** [comment]
> Do you only need this in the session endpoint? Because your changes only touch that enpoint _[...]_

Yes, only here, thanks.

> and also does not change the UI

Indeed. For Hedera verification, we decided to directly consume `server` and `repository` images. Given that `ui` changes might be too Hedera specific, we decided to keep the `ui` fork with our custom modifications.

In any case, this `dryrun` feature is used the contract verification within https://hashscan.io.

**Review by kuzdogan — APPROVED** [review]
LGTM for this specific case. 

We should add `dryRun` to  all other endpoints when doing #1318

**Pull request merged**